duplex_http_call.go

```diff
@@ -229,10 +234,12 @@ func (d *duplexHTTPCall) BlockUntilResponseReady() error {
 // ensureRequestMade sends the request headers and starts the response stream.
 // It is not safe to call this concurrently. Write and CloseWrite call this but
 // ensure that they're not called concurrently.
-func (d *duplexHTTPCall) ensureRequestMade() {
+func (d *duplexHTTPCall) ensureRequestMade() (isFirst bool) {
 	d.sendRequestOnce.Do(func() {
+		isFirst = true
```

Comment:
Eek, this is kinda weird, relying on only one closure over the variable actually mutating it.

I know I suggested _not_ changing the `Once` to an `atomic.Bool` in a past PR, but that was because in that PR it was being changed unnecessarily and felt like "noise" in the diff.. But here, it's actually the perfect change to make since it would make this construct more clear.

Reply:
Switched to atomic.